### PR TITLE
#931 query key factories 2 - He ate 4 things every day and didn't get lung cancer -- [2]

### LINKED ISSUE #931: query key factories for maintainability
Currently we're *generally* inlining query keys.

I've already come up against a couple of bugs where I've miss copy pasted a query key and/or mis-typed.

It would be good to have a strong pattern for query keys to:

- Avoid mistyping / using the wrong key
- Easily able to determine at what level in the hierarchy we can invalidate, at
- Easily use the query keys

One way is to use a global query key file/folder, with all the query keys, so that we can easily view all query keys. This would also stop accidentally using a query key the same as another feature.

However, I think colocation next to where the queries are used is generally better all round.

I am thinking to have something like this in the `useXXApi` hook:

```
const keys = {
  base: () => [ 'requisition', storeId ],
  detail: (id) => [ ...keys.base(), id ],
  list: () => [ ...keys.base(), 'list' ],
  listWithParam: (params) => [...keys.list(),  params ],
}
```

Then when we use the api..

`useQuery(api.keys.detail(id), () => api.get.byId(id))`

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #931 

- Oh how I love a nice good PR apparently
- Think it's relatively straight forward. 
- There are some differences in some of the similar things like query params and the keys in general. As much as it pains me to duplicate so much (and not create the DREADED .. abstract factory?! for query keys) and not create a common shared type I think it's better off for now just being duplicated and waiting a bit longer to properly see the pattern emerge
- But.. in general, putting these `useXApi` and actually just the hooks in general behind some interfaces would be nice 🤷

### CODE REVIEW

File: packages/invoices/src/InboundShipment/api/hooks.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ export const useInboundApi = () => {
   const keys = {
     base: () => ['inbound'] as const,
     detail: (id: string) => [...keys.base(), storeId, id] as const,
-    list: (params: ListParams) =>
-      [...keys.base(), storeId, 'list', params] as const,
+    list: () => [...keys.base(), storeId, 'list'] as const,
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/api/hooks.ts:36]: missed one in this file for deleting inbounds which I've updated in a later PR

**Review by mark-prins — APPROVED** [review]
👍  wow! thanks - that's great. 
also super pleased about the lung cancer

File: packages/common/src/authentication/api/hooks/useGetRefreshToken.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import { useAuthApi } from './useAuthApi';
 import { useQuery } from 'react-query';
-import { AuthQueries } from '../api';
 import { COOKIE_LIFETIME_MINUTES } from '../../AuthContext';
 
 export const useGetRefreshToken = (token: string) => {
   const api = useAuthApi();
-  return useQuery(['refresh-token', token], AuthQueries.get.refreshToken(api), {
+  return useQuery(api.keys.refresh(token), () => api.get.refreshToken(), {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/common/src/authentication/api/hooks/useGetRefreshToken.ts]: Nope. Nice, thanks

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 3]
```diff
@@ -37,23 +37,25 @@ import {
...
   const { invoiceNumber = '' } = useParams();
   return invoiceNumber;
 };
 
-export const useOutboundDetailQueryKey = (): ['invoice', string] => {
-  const outboundNumber = useOutboundNumber();
-  return ['invoice', outboundNumber];
-};
-
 export const useOutbounds = () => {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:59]: good catch 😁

**Pull request merged**